Cursor.md

```diff
@@ -0,0 +1,1712 @@
+# Cursor AI Development Guide
+
+This is a comprehensive guide for AI assistants working on this NestJS boilerplate project. It consolidates all patterns, conventions, and step-by-step instructions needed to solve development tasks.
+
+---
+
+## Table of Contents
+
+1. [Project Overview & Architecture](#project-overview--architecture)
+2. [Quick Reference Checklist](#quick-reference-checklist)
+3. [Creating a New API Resource (Complete Guide)](#creating-a-new-api-resource-complete-guide)
+4. [Entity Creation Patterns](#entity-creation-patterns)
+5. [Migration Management](#migration-management)
+6. [Common Patterns & Conventions](#common-patterns--conventions)
+7. [Module-Specific Guidelines](#module-specific-guidelines)
+8. [Authentication & Security](#authentication--security)
+9. [Development Setup](#development-setup)
+10. [Environment Variables Reference](#environment-variables-reference)
+
+---
+
+## Project Overview & Architecture
+
+### Architecture Pattern
+
+This NestJS application uses a **micro backend architecture** where each functional domain is organized as a separate module (indicated by `app-*` prefixes).
+
+### Directory Structure
+
+```
+src/
+├── app-analytics/      # Analytics tracking and reporting module
+├── app-api/            # API key management and client resources module
+├── app-auth/           # Authentication and authorization module
+├── app-budget/         # User budget management module
+├── app-documents/      # Document management module (files, attachments, groups)
+├── app-jobs/           # Background jobs and scheduled tasks module
+├── common/             # Shared utilities, services, guards, filters
+├── entities/           # TypeORM entity definitions
+├── middleware/         # Express middleware
+├── migrations/         # Database migrations
+├── templates/          # Email templates (Handlebars)
+├── app.controller.ts   # Root application controller
+├── app.module.ts       # Root application module
+├── main.ts             # Application entry point
+└── ormconfig.ts        # TypeORM configuration
+```
+
+### Module Organization
+
+- **App Modules** (`app-*` folders): Feature-based modules that group related functionality
+- **Common** (`common/`): Shared utilities, services, guards, filters, and DTOs used across modules
+- **Entities** (`entities/`): TypeORM database entity definitions
```

Comment:
grouping the entities in a single place and not in every nest module helps separate the bindings between modules